test/unit/int_algolia/scripts/algolia/model/algoliaLocalizedProduct.test.js

```diff
@@ -64,6 +64,76 @@ jest.mock('*/cartridge/scripts/algolia/customization/productModelCustomizer', ()
     return jest.requireActual('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/customization/productModelCustomizer');
 }, {virtual: true});
 
+// Mock for storeAvailability tests
+jest.mock('dw/object/SystemObjectMgr', () => {
+    return {
+        getAllSystemObjects: jest.fn().mockImplementation((objType) => {
+            if (objType === 'Store') {
+                const storeIterator = {
+                    hasNext: function() {
```

Comment:
I just noticed that: we have a mock for `Collection` which already implements an iterator, let's use it: https://github.com/algolia/algoliasearch-sfcc-b2c/blob/15481bb26375883fbdf5d335a9136d546cfb4c29/test/mocks/dw/util/Collection.js